relbench/datasets/f1.py

```diff
@@ -0,0 +1,203 @@
+        constructor_results = constructor_results.merge(
+            races[["raceId", "date"]], on="raceId", how="left"
+        )
+        constructor_standings = constructor_standings.merge(
+            races[["raceId", "date"]], on="raceId", how="left"
+        )
+
+        qualifying = qualifying.merge(
+            races[["raceId", "date"]], on="raceId", how="left"
+        )
+
+        # subtract a day from the date to account for the fact
+        # that the qualifying time is the day before the main race
+        qualifying["date"] = qualifying["date"] - pd.Timedelta(days=1)
+
+        tables = {}
+
+        tables["races"] = Table(
+            df=pd.DataFrame(races),
+            fkey_col_to_pkey_table={
+                "circuitId": "circuits",
+            },
+            pkey_col="raceId",
+            time_col="date",
+        )
+
+        tables["circuits"] = Table(
+            df=pd.DataFrame(circuits),
+            fkey_col_to_pkey_table={},
+            pkey_col="circuitId",
+            time_col=None,
+        )
+
+        tables["drivers"] = Table(
+            df=pd.DataFrame(drivers),
+            fkey_col_to_pkey_table={},
+            pkey_col="driverId",
+            time_col=None,
+        )
+
+        tables["results"] = Table(
+            df=pd.DataFrame(results),
+            fkey_col_to_pkey_table={"raceId": "races", "driverId": "drivers"},
+            pkey_col="resultId",
+            time_col="date",
+        )
+
+        tables["standings"] = Table(
+            df=pd.DataFrame(standings),
+            fkey_col_to_pkey_table={"raceId": "races", "driverId": "drivers"},
+            pkey_col="driverStandingsId",
+            time_col="date",
+        )
+
+        tables["constructors"] = Table(
+            df=pd.DataFrame(constructors),
+            fkey_col_to_pkey_table={},
+            pkey_col="constructorId",
+            time_col=None,
+        )
+
+        tables["constructor_results"] = Table(
+            df=pd.DataFrame(constructor_results),
+            fkey_col_to_pkey_table={"raceId": "races", "constructorId": "constructors"},
+            pkey_col="constructorResultsId",
+            time_col="date",
+        )
+
+        tables["constructor_standings"] = Table(
+            df=pd.DataFrame(constructor_standings),
+            fkey_col_to_pkey_table={"raceId": "races", "constructorId": "constructors"},
+            pkey_col="constructorStandingsId",
+            time_col="date",
+        )
+
+        tables["qualifying"] = Table(
+            df=pd.DataFrame(qualifying),
+            fkey_col_to_pkey_table={
+                "raceId": "races",
+                "driverId": "drivers",
+                "constructorId": "constructors",
+            },
+            pkey_col="qualifyId",
+            time_col="date",
+        )
+
+        db = Database(tables)
+        # only use data after start_timestamp
+        db = db.after(self.start_timestamp)
+
+        return Database(tables)
```

Comment:
Does this `after` actually change the tables in-place? If so, can we add a test?

Reply:
I decided to just remove this `after` method. Currently it is in-effect unused since `self.start_timestamp` is set to the earliest date in the database. I had been testing some alternative cutoff times due to distribution shift, but in the end found this not necessary to train a performant GNN.